articles/quickstart/native/ionic-react/01-login.md

```diff
@@ -0,0 +1,262 @@
+:::
+
+Add a new file `LoginButton.tsx` with the following code:
+
+```js
+import { useAuth0 } from "@auth0/auth0-react";
+import { Browser } from "@capacitor/browser";
+import { IonButton } from "@ionic/react";
+
+const LoginButton: React.FC = () => {
+  const { buildAuthorizeUrl } = useAuth0();
+
+  const login = async () => {
+    // Ask auth0-react to build the login URL
+    const url = await buildAuthorizeUrl();
+
+    // Redirect using Capacitor's Browser plugin
+    await Browser.open({ url });
+  };
+
+  return <IonButton onClick={login}>Log in</IonButton>;
+};
+
+export default LoginButton;
+```
+
+<%= include('../../_includes/_auth0-react-classes-info.md') %>
+
+### Handling the callback
+
+Once a user has logged in using the Universal Login Page, they will be redirected back to your app using a URL with a custom URL scheme. The `appUrlOpen` event must be handled within your app, where `handleRedirectCallback` can be called to initialize the authentication state within the SDK.
+
+Add a handler to your `App` component that looks like the following:
+
+```js
+const { handleRedirectCallback } = useAuth0();
+
+useEffect(() => {
+  // Handle the 'appUrlOpen' event and call `handleRedirectCallback`
+  CapApp.addListener("appUrlOpen", async ({ url }) => {
+      if (
+        url.includes("state") &&
+        (url.includes("code") || url.includes("error"))
+      ) {
+        await handleRedirectCallback(url);
+      }
+      // Ignored on Android
+      await Browser.close();
+  });
+}, [handleRedirectCallback]);
+```
+
+:::panel Checkpoint
+Add the `LoginButton` component to your application, as well as the handler for the "appUrlOpen" event to your `App` component. When you click the login button, verify that your application redirects you to the Auth0 Universal Login Page and that you can now log in or sign up using a username and password or a social provider.
+
+Once that's complete, verify that Auth0 redirects you back to your application.
+:::
+
+## Add Logout to Your Application
+
+Now that you can log in, you need [a way to log out](https://auth0.com/docs/logout/guides/logout-auth0). To do this, create a logout button using the `logout` function from the `useAuth0()` hook. Executing `logout` redirects your users to your [Auth0 logout endpoint](https://auth0.com/docs/api/authentication?javascript#logout) (`https://YOUR_DOMAIN/v2/logout`) and then immediately redirects them to your application.
+
+Similar to `buildAuthorizeUrl`, there is an equivalent method `buildLogoutUrl` that can be opened using Capacitor's Browser plugin. However, you will also need to pass a parameter to `logout` that will clear the SDK's state but not perform the redirect to the logout endpoint, as you want to instead do the redirect with the Browser plugin.
+
+Create a new file `LogoutButton.tsx` and add the following code to the file. Then, add the `LogoutButton` component to your app.
+
+```js
```

Comment:
Should this have a potential file name?
```js
// ./src/components/LoginButton.tsx
```

Reply:
Possible, but I've tried to be less descriptive about where these things go in their app as that's kind of up to them to decide. They might not have a `components` folder for example. I was even loathe to mention a filename at all and just mention "Create a new component with the following code", as again the assumption is that the developer would know where to put this.